app/components/meeting/edit-meeting-modal.js

```diff
@@ -256,6 +264,22 @@ export default class MeetingEditMeetingComponent extends Component {
           }
         }
       }
+
+      // Update secretary in minutes
+      const minutes = yield this.args.meeting.minutes;
+      if (minutes) {
+        const piecePart = yield this.store.queryOne('piece-part', {
+          'filter[:has-no:next-piece-part]': true,
+          'filter[minutes][:id:]': minutes.id,
+        });
+        const newValue = replaceSecretary(piecePart.value, this.secretary.person.get('fullName'));
+        piecePart.value = newValue;
+        yield piecePart.save();
+        yield this.decisionReportGeneration.generateReplacementReport.perform(
+          minutes,
+          'minutes'
+        );
+      }
```

Comment:
**issue**: re-generating the minutes should occur after the meeting is saved

If we re-generate the minutes here, the secretary linked to the meeting is still the old one, which makes the bottom-right secretary name wrong. We should wait for the meeting to be saved before re-generating the report. This code (and check) should be moved lower in the function.

Reply:
Hmm I guess you're right, however, this seems to be working so I'm not really sure how that is possible. Will investigate.

Reply:
Ah, I just noticed there should also be a signature at the bottom of the page, this is not appearing on my end.

Reply:
fixed in [1cd83a7](https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1887/commits/1cd83a744dfb82d8ebf1b4246e3d8161e4a01c0a) (replace after save)